### ISSUE TITLE
[Question] How to install gcr-cleaner into Kubernetes

### ISSUE THREAD

**Issue by KosShutenko** [GoogleCloudPlatform/gcr-cleaner#126]
### TL;DR

I've tried to install gcr-cleaner into Kubernetes as a CronJob. What I did:
1) Configured world identity by the official tutorial: https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity
2) Added the following roles for GCP IAM ServiceAccount:
    - roles/artifactregistry.repoAdmin
    - roles/browser
3) Added add-iam-policy-binding to my repo from your tutorial: https://github.com/GoogleCloudPlatform/gcr-cleaner#artifact-registry
4) I've created CronJob like this:
```
---
apiVersion: batch/v1
kind: CronJob
metadata:
    name: gcr-cleaner
    namespace: gcr-cleaner
spec:
    suspend: true
    successfulJobsHistoryLimit: 1
    failedJobsHistoryLimit: 1
    schedule: 0 0 * * *
    jobTemplate:
      spec:
        template:
          spec:
            serviceAccountName: gcr-cleaner
            restartPolicy: OnFailure
            containers:
              - name: gcr-cleaner
                image: europe-docker.pkg.dev/gcr-cleaner/gcr-cleaner/gcr-cleaner-cli
                imagePullPolicy: IfNotPresent
                env:
                  - name: GCRCLEANER_LOG
                    value: debug
                args:
                  - "-repo=europe-west2-docker.pkg.dev/my-project/my-docker-repo"
                  - "-recursive"
                  - "-keep=20"
                  - "-tag-filter-any=\"^.*$\""
```

But gcr-cleaner doesn't see my repos. See log below.
In that time I can run gcr-cleaner locally from my PC with the following command and it works fine:
```
docker run -v "${HOME}/.config/gcloud:/.config/gcloud" -it --rm europe-docker.pkg.dev/gcr-cleaner/gcr-cleaner/gcr-cleaner-cli -repo=europe-west2-docker.pkg.dev/my-project/my-docker-repo -recursive -dry-run=true -keep=5 -tag-filter-any="^.*$"
```

Looks like I need add additional permissions. But don't see any errors/exceptions in debug log. Can you advice something?

### Expected behavior

I expect that see in logs something like:
```
Deleting refs older than 2023-01-04T15:41:39Z on 2 repo(s)...

europe-west2-docker.pkg.dev/my-project/my-docker-repo/folder1
  ✗ no refs were deleted

europe-west2-docker.pkg.dev/my-project/my-docker-repo/folder2
  ✗ no refs were deleted
```

### Observed behavior

I see the following log:
```
Deleting refs older than 2023-01-04T16:14:33Z on 0 repo(s)...
```

### Debug log output

```text
{"message":"cli is starting","severity":"DEBUG","time":"2023-01-04T16:14:33Z","version":"gcr-cleaner-cli 0.11.0 (59b043bb4c447001fd263f9ffe3b464042e54364, linux/amd64)"}
{"message":"gathering child repositories recursively","severity":"DEBUG","time":"2023-01-04T16:14:33Z"}
{"message":"finding all child repositories","roots":["europe-west2-docker.pkg.dev/my-project/my-docker-repo"],"severity":"DEBUG","time":"2023-01-04T16:14:33Z"}
{"message":"listing child repositories for registry","registry":"europe-west2-docker.pkg.dev","severity":"DEBUG","time":"2023-01-04T16:14:33Z"}
{"message":"found child repositories for registry","registry":"europe-west2-docker.pkg.dev","repos":null,"severity":"DEBUG","time":"2023-01-04T16:14:34Z"}
{"in":["europe-west2-docker.pkg.dev/my-project/my-docker-repo"],"message":"recursively listed child repositories","out":[],"severity":"DEBUG","time":"2023-01-04T16:14:34Z"}
{"message":"cli finished","severity":"DEBUG","time":"2023-01-04T16:14:34Z"}
Deleting refs older than 2023-01-04T16:14:33Z on 0 repo(s)...
2023-01-04T16:14:34.587391057Z
```
```


### Additional information

_No response_

**Comment by sethvargo** [comment]
Hi @KosShutenko - thank you for opening an issue and supplying all the debug information. This looks like an issue with Workload Identity. Could you try using a `gcloud` container and see what happens:

```yaml
---
apiVersion: batch/v1
kind: CronJob
metadata:
    name: gcr-cleaner
    namespace: gcr-cleaner
spec:
    suspend: true
    successfulJobsHistoryLimit: 1
    failedJobsHistoryLimit: 1
    schedule: 0 0 * * *
    jobTemplate:
      spec:
        template:
          spec:
            serviceAccountName: gcr-cleaner
            restartPolicy: OnFailure
            containers:
              - name: gcr-cleaner
                image: google/cloud-sdk:latest
                imagePullPolicy: Always
                env:
                  - name: GCRCLEANER_LOG
                    value: debug
                args: ["gcloud", "auth", "list"]
```

And provide the output? There's also some other helpful debugging tips in the [Workload Identity documentation](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#verify_the_setup).

**Comment by KosShutenko** [comment]
> Hi @KosShutenko - thank you for opening an issue and supplying all the debug information. This looks like an issue with Workload Identity. Could you try using a `gcloud` container and see what happens:
> 
> ```yaml
> ---
> apiVersion: batch/v1
> kind: CronJob
> metadata:
>     name: gcr-cleaner
>     namespace: gcr-cleaner
> spec:
>     suspend: true
>     successfulJobsHistoryLimit: 1
>     failedJobsHistoryLimit: 1
>     schedule: 0 0 * * *
>     jobTemplate:
>       spec:
>         template:
>           spec:
>             serviceAccountName: gcr-cleaner
>             restartPolicy: OnFailure
>             containers:
>               - name: gcr-cleaner
>                 image: google/cloud-sdk:latest
>                 imagePullPolicy: Always
>                 env:
>                   - name: GCRCLEANER_LOG
>                     value: debug
>                 args: ["gcloud", "auth", "list"]
> ```
> 
> And provide the output? There's also some other helpful debugging tips in the [Workload Identity documentation](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#verify_the_setup).

Tried. he following Cronjob:
```
---
apiVersion: batch/v1
kind: CronJob
metadata:
    name: gcr-cleaner
    namespace: gcr-cleaner
spec:
    suspend: true
    successfulJobsHistoryLimit: 1
    failedJobsHistoryLimit: 1
    schedule: 0 0 * * *
    jobTemplate:
      spec:
        template:
          spec:
            serviceAccountName: gcr-cleaner
            restartPolicy: OnFailure
            containers:
              - name: gcr-cleaner
                image: google/cloud-sdk:latest
                imagePullPolicy: Always
                env:
                  - name: GCRCLEANER_LOG
                    value: debug
                args: ["gcloud", "auth", "list"]
```

and got the following log:
```
                       Credentialed Accounts
ACTIVE  ACCOUNT

To set the active account, run:
    $ gcloud config set account `ACCOUNT`

*       gcr-admin@my-project.iam.gserviceaccount.com
```

Also, I've tried to test Workload identity via: https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#verify_the_setup
and got the following:
```
root@workload-identity-test:/# curl -H "Metadata-Flavor: Google" http://169.254.169.254/computeMetadata/v1/instance/service-accounts/default/email
gcr-admin@my-project.iam.gserviceaccount.com
root@workload-identity-test:/# 
```

**Comment by sethvargo** [comment]
Interesting - that confirms that the service account is configured correctly.

**Comment by KosShutenko** [comment]
Yes, looks like I need additional permissions. But, unfortunately, I don't know which.

**Comment by ahsan-raza** [comment]
@KosShutenko did you find the resolution to this? I am facing the same problem

**Comment by ahsan-raza** [comment]
I was able to resolve this. It was issue of service account permission. I assigned the permission to different repository and was trying to delete the images from the different repository.
